update to new docs location (#330)
fluxcd/website: README.md

> > **Note**: The sources for some of Flux's documentation are housed in the other Flux repositories within <https://github.com/fluxcd>. Issues and pull requests for this documentation should be registered at that repos.

Reviewer: ```suggestion
> **Note**: The sources for some of Flux's documentation are housed in other repositories within <https://github.com/fluxcd>. Documentation issues and pull requests should be made against those repos.
```

>
> Project       | Docs Site                                 | Github Source
> ------------- | ------------------------------------------| -------------

> > Flux v1       | <https://fluxcd.io/legacy/flux>           | <https://github.com/fluxcd/website>

Reviewer: I would list this as Flux v2, Flagger, Flux v1, Helm Operator v1.

> Helm Operator | <https://fluxcd.io/legacy/helm-operator/> | <https://github.com/fluxcd/website>
> Flagger       | <https://docs.flagger.app>                | <https://github.com/fluxcd/flagger>
> Flux v2       | <https://fluxcd.io/docs>                  | <https://github.com/fluxcd/website>
>

> > We are in the process of moving Flagger docs into this repository as well.

Reviewer: We are not in the process of moving Flagger docs, maybe next year, please remove this.
